src/main/java/com/mraof/minestuck/entity/item/GristEntity.java

```diff
@@ -199,27 +243,67 @@ protected void readAdditionalSaveData(CompoundTag compound)
 			this.gristType = GristType.read(compound, "Type");
 	}
 	
+	public int shaderAlpha = 0;
+	
 	/**
 	 * Called by a player entity when they collide with an entity
+	 * dectects collision with player entity and
 	 */
 	@Override
 	public void playerTouch(Player entityIn)
 	{
-		if(this.level.isClientSide?ClientEditHandler.isActive():ServerEditHandler.getData(entityIn) != null)
-			return;
+		if(this.level.isClientSide ? ClientEditHandler.isActive() : ServerEditHandler.getData(entityIn) != null)
+			return; //checks if player is in edit mode. returns nothing and doesn't allow the entities to touch.
 		
-		if (!this.level.isClientSide && !(entityIn instanceof FakePlayer))
+		if(entityIn instanceof ServerPlayer)
 		{
-			consumeGrist(IdentifierHandler.encode(entityIn), true);
+			Session playerSession = SessionHandler.get(level).
+					getPlayerSession(IdentifierHandler.encode(entityIn));
+			
+			long playerGristAmount;
+			long rung;
+			
+			playerGristAmount = PlayerSavedData.getData((ServerPlayer) entityIn).getGristCache().getGrist(gristType);
+			rung = PlayerSavedData.getData((ServerPlayer) entityIn).getEcheladder().getRung();
+			
+			int gristCap = GristHelper.rungGrist[(int) rung];
+			int gutterCap = 0;
+			long gutterTotal = 0;
+			
+			if(playerSession != null)
+			{
+				gutterTotal = playerSession.getGristGutter().getGutterTotal();
+				gutterCap = GristGutter.GUTTER_CAPACITY;
+			}
+			
+			long gutterRoom = gutterCap - gutterTotal;
+			long cacheRoom = gristCap - playerGristAmount;
+			
+			boolean hasRoom = (gutterRoom + cacheRoom) >= (gristValue);
+			
+			if(hasRoom)
+			{
+				consumeGrist(IdentifierHandler.encode(entityIn), true);
+			}
+			else
+			{
+				this.animation = Animation.REJECT;
+				GristEntityPacket packet = GristEntityPacket.createPacket(this, animation);
+				MSPacketHandler.sendToTracking(packet, this);
+				shaderAlpha = 255;
+			}
 		}
 	}
 	
+	
 	public void consumeGrist(PlayerIdentifier identifier, boolean sound)
 	{
+		
```

Comment:
These extra line breaks probably shouldn't be here.

Reply:
See my comment on EcheladderScreen

Reply:
done now die(jk i love you dweb you're my fave)